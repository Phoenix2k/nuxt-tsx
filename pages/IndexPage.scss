.container {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 0 auto;
  min-height: 100vh;
  text-align: center;
}

.links {
  padding-top: 1rem;
}

.subtitle {
  color: #526488;
  font-size: 2.6rem;
  font-weight: 300;
  padding-bottom: 1rem;
}

.title {
  color: #35495e;
  display: block;
  font-family: 'Quicksand', 'Source Sans Pro', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    'Helvetica Neue', Arial, sans-serif;
  font-size: 6rem;
  font-weight: 300;
  letter-spacing: 0.015em;
}
