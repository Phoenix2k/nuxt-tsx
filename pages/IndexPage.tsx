import Vue, { VNode } from 'vue';
import Logo from '@/components/Logo';
import './IndexPage.scss';

export default Vue.extend({
  components: {
    Logo
  },
  render(): VNode {
    return (
      <div class="container">
        <div>
          <Logo />
          <h1 class="title" itemprop="title">
            nuxt-tsx
          </h1>
          <h2 class="subtitle">Nuxt.js build with TypeScript and TSX support</h2>
          <div class="links">
            <nuxt-link class="button button--grey" to="/about">
              About page
            </nuxt-link>
            <a class="button button--green" href="https://nuxtjs.org/" target="_blank">
              Documentation
            </a>
            <a class="button button--grey" href="https://github.com/nuxt/nuxt.js" target="_blank">
              GitHub
            </a>
          </div>
        </div>
      </div>
    );
  }
});
