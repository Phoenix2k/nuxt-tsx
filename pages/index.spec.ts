/// <reference types="jest" />

import { mount, RouterLinkStub } from '@vue/test-utils';
import IndexPage from '.';

const options = {
  stubs: {
    NuxtLink: RouterLinkStub
  }
};

const wrapper = mount(IndexPage, options);

describe('Index page', () => {
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
