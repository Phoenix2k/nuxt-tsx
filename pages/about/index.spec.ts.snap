// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders correctly 1`] = `
<div
  class="container"
>
  <div>
    <h1
      class="title"
    >
      About
    </h1>
    <h2
      class="subtitle"
    >
      This is the about page.
    </h2>
    <div
      class="links"
    >
      <a
        class="button button--grey"
      >
        Home page
      </a>
    </div>
  </div>
</div>
`;
